File: project-in-dotnet/Vehicle_Configurator_Backend/Vehicle_Configurator/Domain/Entities/InvoiceDetail.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Vehicle_Configurator.Domain.Entities
{
    [Table("invoice_detail")]
    public class InvoiceDetail
    {
        [Key]
        [Column("inv_dtl_id")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int InvDtlId { get; set; }

        [Column("inv_id")]
        [ForeignKey("Invoice")]
        public int InvId { get; set; }

        public InvoiceHeader Invoice { get; set; }

        [Column("comp_id")]
        [ForeignKey("Component")]
        public int CompId { get; set; }
        public Component Component { get; set; }
    }
}

File: project-in-java/backend/Vehicle_Configurator-main/src/main/java/com/example/entity/Invoice_detail.java
package com.example.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToOne;
import lombok.Data;

@Entity
@Data
public class Invoice_detail {
	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int invDtl_id;
	
	@ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "inv_id", referencedColumnName = "inv_id")
    private Invoice_header invoice;
	
	@ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "comp_id", referencedColumnName = "comp_id")
	@JsonIgnore
    private Component component;
}
